### PR TITLE
Update raft for compatibility with the latest cuco

### PR DESCRIPTION
**Pull request by PointKernel**
This PR updates raft to make it compatible with the latest cuco.

Depends on https://github.com/rapidsai/rapids-cmake/pull/526

CMake changes will be reverted once https://github.com/rapidsai/rapids-cmake/pull/526 is merged.

### CODE REVIEW

File: cpp/include/raft/sparse/distance/detail/coo_spmv_strategies/hash_strategy.cuh [hunk 1]
```diff
@@ -43,11 +43,11 @@ namespace detail {
...
 class hash_strategy : public coo_spmv_strategy<value_idx, value_t, tpb> {
  public:
-  using insert_type =
-    typename cuco::static_map<value_idx, value_t, cuda::thread_scope_block>::device_mutable_view;
+  using insert_type = typename cuco::legacy::
+    static_map<value_idx, value_t, cuda::thread_scope_block>::device_mutable_view;
   using smem_type = typename insert_type::slot_type*;
   using find_type =
-    typename cuco::static_map<value_idx, value_t, cuda::thread_scope_block>::device_view;
+    typename cuco::legacy::static_map<value_idx, value_t, cuda::thread_scope_block>::device_view;
```
[the remark this answers is not in the document]
Comment (PointKernel) [reply cpp/include/raft/sparse/distance/detail/coo_spmv_strategies/hash_strategy.cuh:50]: >  is there another similar API we can migrate this to

Yes, we added a more performant [new map](https://github.com/NVIDIA/cuCollections/blob/a992e356c30698224ee89c2e29029425db72825b/include/cuco/static_map.cuh#L88-L97) and it's ready to use. I can put up a PR afterward for the map migration.

**Review by bdice — APPROVED** [review]

**Comment by PointKernel** [comment]
This is ready to go once CI passes. Note the CI will be blocked until this PR gets merged. 

Thanks!

**Comment by PointKernel** [comment]
Can someone please rerun the failed CI task? I didn't see a similar issue in previous CI checks thus maybe it's a random CI runtime error?

**Pull request merged**